src/binary/raw_binary_reader.rs

```diff
@@ -379,36 +389,53 @@ impl<R: IonDataSource> RawReader for RawBinaryReader<R> {
         self.cursor.value.is_null
     }
 
-    fn annotations(&self) -> &[RawSymbolToken] {
+    fn annotations<'a>(&'a self) -> Box<dyn Iterator<Item = IonResult<Self::Symbol>> + 'a> {
```

Comment:
PR tour: when [GATs land](https://blog.rust-lang.org/2021/08/03/GATs-stabilization-push.html) in May/June, we can replace this with an associated type whose lifetime is bound to `'a`. Then it won't require heap allocation and virtual dispatch.